- Description: zfcp: Always initialize host_scribble field.
  Symptom:     Addressing exception, system hangs
  Problem:     The zfcp driver sometimes passes commands back to the SCSI
               midlayer where the host_scribble field is not set to NULL.
               When such a command is retried this might lead to an addressing
               exception since the host_scribble field points then to an
               already freed memory area.
  Solution:    Initialize host_scribble to NULL in queuecommand function.
  Problem-ID:  -
  Archs:       s390-31, s390-64

--- linux-2.5/drivers/s390/scsi/zfcp_scsi.c	19 Apr 2004 10:49:37 -0000	1.59.2.1
+++ linux-2.5/drivers/s390/scsi/zfcp_scsi.c	23 Apr 2004 14:36:18 -0000	1.59.2.2
@@ -31,7 +31,7 @@
 #define ZFCP_LOG_AREA			ZFCP_LOG_AREA_SCSI
 
 /* this drivers version (do not edit !!! generated and updated by cvs) */
-#define ZFCP_SCSI_REVISION "$Revision: 1.59.2.1 $"
+#define ZFCP_SCSI_REVISION "$Revision: 1.59.2.2 $"
 
 #include <linux/blkdev.h>
 
@@ -345,7 +345,7 @@
 
 	/* reset the status for this request */
 	scpnt->result = 0;
-	/* save address of mid layer call back function */
+	scpnt->host_scribble = NULL;
 	scpnt->scsi_done = done;
 
 	/*

